radixdlt/src/main/java/com/radixdlt/consensus/liveness/ProposalGenerator.java

```diff
@@ -0,0 +1,37 @@
+package com.radixdlt.consensus.liveness;
+
+import com.google.common.collect.Sets;
+import com.radixdlt.common.Atom;
+import com.radixdlt.consensus.QuorumCertificate;
+import com.radixdlt.consensus.Round;
+import com.radixdlt.consensus.Vertex;
+import com.radixdlt.consensus.VertexStore;
+import com.radixdlt.mempool.Mempool;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Logic for generating new proposals
+ */
+public final class ProposalGenerator {
+	private final Mempool mempool;
+	private final VertexStore vertexStore;
+
+	@com.google.inject.Inject
```

Comment:
Would be better to use @Inject rather than fully qualified name.

Would also be better to use javax.inject version, as this is not specific to guice.